agents/codex/deployment/build-mlflow/setup-mlflow.sh

```
@@ -0,0 +1,122 @@
+{
+  "trackingUri": "${MLFLOW_TRACKING_URI}",
+  "experimentId": "${exp_id}"
+}
+TRACING_JSON
+        log_info "mlflow-tracing.json written (experiment ${exp_id})"
+
+        # Ensure notify hook is in user-level config.toml
+        # Uses mlflow-codex-hook.sh wrapper so the SA token is loaded even when
+        # codex is launched via oc exec (which bypasses .bashrc).
+        if ! grep -q '^notify = ' "${CODEX_HOME}/config.toml" 2>/dev/null; then
+            echo '' >> "${CODEX_HOME}/config.toml"
```

Shellcheck (0.11.0)

[style] Consider using { cmd1; cmd2; } >> file instead of individual redirects.

(SC2129)